frontend/src/routes/Applications/CreateApplication/Subscription/controlData/ControlDataPlacement.js

```diff
@@ -50,142 +65,149 @@ export const loadExistingPlacementRules = () => {
-
-            _.set(onlineControl, 'type', 'checkbox')
-            _.set(onlineControl, 'disabled', false)
-            _.set(clusterSelectorControl, 'type', 'custom')
-            _.set(localClusterControl, 'type', 'checkbox')
-        }
-
-        //reset all values
-        _.set(localClusterControl, 'active', false)
-        _.set(onlineControl, 'active', false)
-        if (clusterSelectorControl.active) {
-            clusterSelectorControl.active.clusterLabelsListID = 1
-            delete clusterSelectorControl.active.clusterLabelsList
-            clusterSelectorControl.active.clusterLabelsList = [
-                { id: 0, labelName: '', labelValue: '', validValue: false },
-            ]
-            clusterSelectorControl.active.mode = true
-            delete clusterSelectorControl.showData
-        }
-        const availablePlacementControl = existingRuleControl.available
-        if (!availablePlacementControl.includes(existingRuleControl.active)) existingRuleControl.active = ''
-    })
-    return controlGlobal
-}
-
-export const updatePlacementControlsForLocal = (placementControl) => {
-    const { active, groupControlData } = placementControl
 
-    const onlineControl = getControlByID(groupControlData, onlineClusterCheckbox)
-    const clusterSelectorControl = getControlByID(groupControlData, 'clusterSelector')
+const setAvailableRules = (control, result) => {
+    const { loading } = result
+    const { data, i18n } = result
+    const placementRules = data
+    control.isLoading = false
+    const error = placementRules ? null : result.error
+    if (!control.available) {
+        control.available = []
+        control.availableData = []
+    }
+    if (error || placementRules) {
+        if (error) {
+            control.isFailed = true
+            control.isLoaded = true
+        } else if (placementRules) {
+            control.isLoaded = true
+
+            control.availableInfo = {}
+            const { groupControlData } = control
+            const enableHubSelfManagement = getControlByID(groupControlData, 'enableHubSelfManagement')
+
+            const keyFn = (rule) => {
+                const ruleName = _.get(rule, 'metadata.name', '')
+                const clusterSelector = _.get(rule, 'spec.clusterSelector')
+                const clusterConditions = _.get(rule, 'spec.clusterConditions')
+                let selector = enableHubSelfManagement?.active
+                    ? i18n('creation.app.local.clusters.only', [ruleName])
+                    : unavailable
+                if (clusterSelector?.matchExpressions?.length > 0) {
+                    if (clusterSelector.matchExpressions[0]?.key !== 'local-cluster') {
+                        const getLabels = () => {
+                            return clusterSelector.matchExpressions
+                                .map(({ key, operator, values }) => {
+                                    return `${key}${operator === 'In' ? '=' : '!='}${values.join(', ')}`
```

Comment:
This does not account for the `Exists` and `DoesNotExist` values for `operator`.

```
properties:
                  matchExpressions:
                    description: matchExpressions is a list of label selector requirements. The requirements are ANDed.
                    items:
                      description: A label selector requirement is a selector that contains values, a key, and an operator that relates the key and values.
                      properties:
                        key:
                          description: key is the label key that the selector applies to.
                          type: string
                        operator:
                          description: operator represents a key's relationship to a set of values. Valid operators are In, NotIn, Exists and DoesNotExist.
                          type: string
                        values:
                          description: values is an array of string values. If the operator is In or NotIn, the values array must be non-empty. If the operator is Exists or DoesNotExist, the values array must be empty. This array is replaced during a strategic merge patch.
```